core/src/main/scala/cats/Foldable.scala

```diff
@@ -214,6 +214,31 @@ import simulacrum.typeclass
         case Right(_) => None
       }
 
+  def collectFirst[A, B](fa: F[A])(pf: PartialFunction[A, B]): Option[B] =
```

Comment:
Is it worth mentioning that a more efficient implementation is often possible?  We do this on `size` and `foldM`, but I suppose it would be easy to get carried away with this.  I don't feel strongly.

Reply:
My battery is going to die before I can run the tests, but I think this might be right:

```scala
    foldRight(fa, Eval.now(Option.empty[B])) { (a, lb) =>
      var result: Eval[Option[B]] = lb
      pf.runWith(b => result = Eval.now(Some(b)))(a)
      result
    }.value
```

Reply:
This is the `TraversableOnce` trick in `foldRight` form.  `TraversableOnce` claims it's faster than `runWith`, but I have not benchmarked it in this context.  Tests pass even after I remove the overrides.  `sentinel` could be moved to a private member of the companion.

```scala
    val sentinel: Function1[A, Any] = new scala.runtime.AbstractFunction1[A, Any]{ def apply(a: A) = this }
    foldRight(fa, Eval.now(Option.empty[B])) { (a, lb) =>
      val x = pf.applyOrElse(a, sentinel)
      if (x.asInstanceOf[AnyRef] ne sentinel) Eval.now(Some(x.asInstanceOf[B]))
      else lb
    }.value
```

Reply:
thanks a lot @rossabaker , I made the change.